### PR TITLE
gh-84461: Add HOSTRUNNER for program to run Python executable

### PR DESCRIPTION
**Pull request by emmatyping**
`HOSTRUNNER` is a program which can be used to run `BUILDPYTHON` for the host platform (for example, `python.js` requires `node`).

Also change depedencies from `build_all` to `all` so that targets which can't build everything (e.g. WASM) can still run `buildbottest` and `pythoninfo`.

cc @tiran

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

**Review by vagobon — APPROVED** [review]

File: configure.ac [hunk 1]
```diff
@@ -1412,6 +1412,17 @@ if test "$cross_compiling" = yes; then
 	RUNSHARED=
 fi
 
+AC_ARG_VAR(HOSTRUNNER, Program to run CPython for the host platform)
+AC_MSG_CHECKING(HOSTRUNNER)
```
Comment (tiran) [inline configure.ac]:
```suggestion
AC_ARG_VAR([HOSTRUNNER], [Program to run CPython for the host platform])
AC_MSG_CHECKING([HOSTRUNNER])
```

File: configure.ac [hunk 2]
```diff
@@ -1412,6 +1412,17 @@ if test "$cross_compiling" = yes; then
...
+if test -z "$HOSTRUNNER"
+then
+	AS_CASE([$ac_sys_system/$ac_sys_emscripten_target],
+  [Emscripten/node*], [HOSTRUNNER='node'],
+  [HOSTRUNNER='']
+)
```
Comment (tiran) [inline configure.ac]:
```suggestion
  AS_CASE([$ac_sys_system/$ac_sys_emscripten_target],
    [Emscripten/node*], [HOSTRUNNER='node'],
    [HOSTRUNNER='']
  )
```

File: configure.ac [hunk 3]
```diff
@@ -1412,6 +1412,17 @@ if test "$cross_compiling" = yes; then
...
+fi
+AC_MSG_RESULT($HOSTRUNNER)
```
Comment (tiran) [inline configure.ac]: How does the code work for you without AC_SUBST?
```suggestion
AC_SUBST([$HOSTRUNNER])
AC_MSG_RESULT([$HOSTRUNNER])
```

**Review by tiran — COMMENTED** [review]
- Add ``[]`` around arguments.
- Fix whitespace / tab
- You need to AC_SUBST variables to make configure substitute the variable in templates.
- Threaded builds need additional arguments to enable experimental threading and shared buffers

```
AS_VAR_IF([enable_wasm_pthreads], [yes], [
  HOSTRUNNER='node --experimental-wasm-threads --experimental-wasm-bulk-memory'
], [
  HOSTRUNNER='node'
])
```

File: configure.ac (hunk 3 above)
Comment (emmatyping) [reply configure.ac]: I think I was using `make ... HOSTRUNNER=...` 😅

File: configure.ac [hunk 4]
```diff
@@ -1412,6 +1412,18 @@ if test "$cross_compiling" = yes; then
 	RUNSHARED=
 fi
 
+AC_ARG_VAR([HOSTRUNNER], [Program to run CPython for the host platform])
+AC_MSG_CHECKING([HOSTRUNNER])
+if test -z "$HOSTRUNNER"
+then
+  AS_CASE([$ac_sys_system/$ac_sys_emscripten_target],
+    [Emscripten/node*], [HOSTRUNNER='node'],
```
Comment (tiran) [inline configure.ac]:
```suggestion
    [Emscripten/node*], [
      AS_VAR_IF([enable_wasm_pthreads], [yes], [
        HOSTRUNNER='node --experimental-wasm-threads --experimental-wasm-bulk-memory'
      ], [
        HOSTRUNNER='node'
      ])
    ],
```

Comment (emmatyping) [reply configure.ac]: Thanks! I was going to do this earlier but ended up in a meeting.

**Review by tiran — APPROVED** [review]

**Pull request merged**